assets/js/modules/analytics-4/components/widgets/TopTrafficSourceDrivingAddToCartWidget.js

```diff
@@ -0,0 +1,217 @@
+import {
+	DATE_RANGE_OFFSET,
+	MODULES_ANALYTICS_4,
+} from '../../datastore/constants';
+import { numFmt } from '../../../../util';
+import { get } from 'lodash';
+import whenActive from '../../../../util/when-active';
+import ConnectGA4CTATileWidget from './ConnectGA4CTATileWidget';
+
+function TopTrafficSourceDrivingAddToCartWidget( { Widget } ) {
+	const dates = useSelect( ( select ) =>
+		select( CORE_USER ).getDateRangeDates( {
+			offsetDays: DATE_RANGE_OFFSET,
+			compare: true,
+		} )
+	);
+
+	const detectedEvents = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getDetectedEvents()
+	);
+	const hasDetectedEvent = detectedEvents?.includes( 'add_to_cart' );
+
+	const totalAddToCartReportOptions = {
+		...dates,
+		metrics: [
+			{
+				name: 'addToCarts',
+			},
+		],
+	};
+
+	const trafficSourceReportOptions = {
+		...dates,
+		dimensions: [ 'sessionDefaultChannelGroup' ],
+		metrics: [
+			{
+				name: 'addToCarts',
+			},
+		],
+		limit: 1,
+		orderBy: 'addToCarts',
+	};
+
+	const totalAddToCartReport = useInViewSelect(
+		( select ) =>
+			hasDetectedEvent
+				? select( MODULES_ANALYTICS_4 ).getReport(
+						totalAddToCartReportOptions
+				  )
+				: undefined,
+		[ hasDetectedEvent, totalAddToCartReportOptions ]
+	);
+
+	const trafficSourceReport = useInViewSelect(
+		( select ) =>
+			hasDetectedEvent
+				? select( MODULES_ANALYTICS_4 ).getReport(
+						trafficSourceReportOptions
+				  )
+				: undefined,
+		[ hasDetectedEvent, trafficSourceReportOptions ]
+	);
+
+	const error = useSelect( ( select ) => {
+		const trafficSourceReportErrors = select(
+			MODULES_ANALYTICS_4
+		).getErrorForSelector( 'getReport', [ trafficSourceReportOptions ] );
+
+		const totalAddToCartReportErrors = select(
+			MODULES_ANALYTICS_4
+		).getErrorForSelector( 'getReport', [ totalAddToCartReportOptions ] );
+
+		if ( trafficSourceReportErrors && totalAddToCartReportErrors ) {
+			return [ trafficSourceReportErrors, totalAddToCartReportErrors ];
+		}
+
+		return (
+			trafficSourceReportErrors || totalAddToCartReportErrors || undefined
+		);
+	} );
+
+	const loading = useSelect( ( select ) =>
+		hasDetectedEvent
+			? ! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+					'getReport',
+					[ totalAddToCartReportOptions ]
+			  ) ||
+			  ! select( MODULES_ANALYTICS_4 ).hasFinishedResolution(
+					'getReport',
+					[ trafficSourceReportOptions ]
+			  )
+			: undefined
+	);
+
+	const makeFilter = ( dateRange, dimensionIndex ) => ( row ) =>
+		get( row, `dimensionValues.${ dimensionIndex }.value` ) === dateRange;
+
+	// Prevents running a filter on `report.rows` which could be undefined.
+	const { rows: totalAddToCartReportRows = [] } = totalAddToCartReport || {};
+	const { rows: trafficSourceReportRows = [] } = trafficSourceReport || {};
+
+	const topTrafficSource =
+		trafficSourceReportRows.filter( makeFilter( 'date_range_0', 1 ) )[ 0 ]
+			?.dimensionValues?.[ 0 ].value || '-';
```

Comment:
I think we need a better UI for cases when there is no traffic source information. Showing `-` and `0% of total add to carts` is completely useless and bad UX. Could you please work with Sigal to figure out what we should display in this case?

Reply:
I pinged @jimmymadon in the group, I can't make changes specific to these ACR widgets, as they incorporate existing layout that is used by few initial key metrics, and they need to be consistent across all widgets. I am checking if this is part of initial design, etc, then I will open a follow up issue to suggest update to this layout, and ping Sigal about. But in the meantime I am collecting more info, and we do not need to hold these issues off for this, as they are behind feature flag and the current zero data state layout is already in use for other core metrics. So this can be improved in a new issue that will target all metrics with this layout

Reply:
New issue opened #9430, will chat with Sigal about it, after other more timely ongoing ACR questions are resolved

Reply:
Thanks, @zutigrm